sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -66,980 +85,305 @@ case class ScalaUDF(
-      lazy val converter6 = CatalystTypeConverters.createToScalaConverter(child6.dataType)
-      lazy val converter7 = CatalystTypeConverters.createToScalaConverter(child7.dataType)
-      lazy val converter8 = CatalystTypeConverters.createToScalaConverter(child8.dataType)
-      lazy val converter9 = CatalystTypeConverters.createToScalaConverter(child9.dataType)
-      lazy val converter10 = CatalystTypeConverters.createToScalaConverter(child10.dataType)
-      lazy val converter11 = CatalystTypeConverters.createToScalaConverter(child11.dataType)
-      lazy val converter12 = CatalystTypeConverters.createToScalaConverter(child12.dataType)
-      lazy val converter13 = CatalystTypeConverters.createToScalaConverter(child13.dataType)
-      lazy val converter14 = CatalystTypeConverters.createToScalaConverter(child14.dataType)
-      lazy val converter15 = CatalystTypeConverters.createToScalaConverter(child15.dataType)
-      lazy val converter16 = CatalystTypeConverters.createToScalaConverter(child16.dataType)
-      lazy val converter17 = CatalystTypeConverters.createToScalaConverter(child17.dataType)
-      lazy val converter18 = CatalystTypeConverters.createToScalaConverter(child18.dataType)
-      lazy val converter19 = CatalystTypeConverters.createToScalaConverter(child19.dataType)
-      lazy val converter20 = CatalystTypeConverters.createToScalaConverter(child20.dataType)
-      lazy val converter21 = CatalystTypeConverters.createToScalaConverter(child21.dataType)
-      (input: InternalRow) => {
-        func(
-          converter0(child0.eval(input)),
-          converter1(child1.eval(input)),
-          converter2(child2.eval(input)),
-          converter3(child3.eval(input)),
-          converter4(child4.eval(input)),
-          converter5(child5.eval(input)),
-          converter6(child6.eval(input)),
-          converter7(child7.eval(input)),
-          converter8(child8.eval(input)),
-          converter9(child9.eval(input)),
-          converter10(child10.eval(input)),
-          converter11(child11.eval(input)),
-          converter12(child12.eval(input)),
-          converter13(child13.eval(input)),
-          converter14(child14.eval(input)),
-          converter15(child15.eval(input)),
-          converter16(child16.eval(input)),
-          converter17(child17.eval(input)),
-          converter18(child18.eval(input)),
-          converter19(child19.eval(input)),
-          converter20(child20.eval(input)),
-          converter21(child21.eval(input)))
-      }
+  private[this] val f = {
+    lazy val inputEncoder: ExpressionEncoder[Row] = RowEncoder(inputSchema)
+    children.size match {
+      case 0 =>
+        val func = function.asInstanceOf[() => Any]
+        (input: InternalRow) => {
+          func()
+        }
+
+      case 1 =>
+        val func = function.asInstanceOf[(Any) => Any]
+        (input: InternalRow) => {
+          val convertedRow: Row = inputEncoder.fromRow(input)
+          func(convertedRow.get(0))
```

Comment:
Could we call this with reflection? Since the interpreted version is not used usually, don't need to worry about the performance.